### PR TITLE
chore: Add `checkbox-state` example

### PR DESCRIPTION
**Pull request by SCasarotto**
This update adds an example for the Checkbox component that uses `useCheckboxState` hook:

Screenshots
image

How to test?

`npm run dev`
Visit http://localhost:3000/examples/checkbox-usecheckboxstate

Does this PR introduce breaking changes?
Nope, just an example

On this one, I wasn't completely sure how to best demonstrate `useCheckboxState`. In your inline docs, you have a few examples:

```
  /**
   * The default value of the checkbox.
   * @default false
   * @example
   * const checkbox = useCheckboxState({ defaultValue: ["Apple", "Orange"] });
   * <Checkbox state={checkbox} value="Apple" />
   * <Checkbox state={checkbox} value="Orange" />
   * <Checkbox state={checkbox} value="Watermelon" />
   */
  defaultValue?: T;
  /**
   * Function that will be called when setting the checkbox `value` state.
   * @example
   * // Uncontrolled example
   * useCheckboxState({ setValue: (value) => console.log(value) });
   * @example
   * // Controlled example
   * const [value, setValue] = useState(false);
   * useCheckboxState({ value, setValue });
   * @example
   * // Externally controlled example
   * function CheckboxGroup({ value, onChange }) {
   *   const checkbox = useCheckboxState({ value, setValue: onChange });
   * }
   */
   setValue?: (value: T) => void;
```

We certainly could do any of those pretty easily. Any thoughts?

### CODE REVIEW

File: packages/ariakit/src/checkbox/__examples__/checkbox-usecheckboxstate/index.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Checkbox, useCheckboxState } from "ariakit/checkbox";
+
+import "./style.css";
+
+export default function Example() {
+  const checked = useCheckboxState({ defaultValue: true });
+
+  return (
```
Comment (diegohaz) [inline packages/ariakit/src/checkbox/__examples__/checkbox-usecheckboxstate/index.tsx]: Nit:
```suggestion
import { Checkbox, useCheckboxState } from "ariakit/checkbox";
import "./style.css";

export default function Example() {
  const checked = useCheckboxState({ defaultValue: true });
  return (
```

**Review by diegohaz — COMMENTED** [review]
Awesome! `useCheckboxState({ defaultValue })` is a good example. I'd just rename it to `checkbox-state` instead of `checkbox-usecheckboxstate`.

**Comment by SCasarotto** [comment]
Name updated

**Review by diegohaz — APPROVED** [review]
LGTM 🚀

**Pull request merged**